strawberry/experimental/pydantic/object_type.py

```diff
@@ -106,24 +107,41 @@ def wrap(cls):
         if not fields_set:
             raise MissingFieldsListError(cls)
 
-        all_model_fields: List[Tuple[str, Any, dataclasses.Field]] = [
-            (
-                name,
-                get_type_for_field(field),
-                StrawberryField(
-                    python_name=field.name,
-                    graphql_name=field.alias if field.has_alias else None,
-                    default=field.default if not field.required else UNSET,
-                    default_factory=(
-                        field.default_factory if field.default_factory else UNSET
-                    ),
-                    type_annotation=get_type_for_field(field),
-                    description=field.field_info.description,
-                ),
-            )
-            for name, field in model_fields.items()
-            if name in fields_set
-        ]
+        all_model_fields: List[Tuple[str, Any, dataclasses.Field]] = []
+
+        for field_name, field in model_fields.items():
+            if field_name in fields_set:
+                default = field.default if not field.required else UNSET
+                default_factory = (
+                    field.default_factory if field.default_factory else UNSET
+                )
+
+                # Handle mutable defaults when making the dataclass by using pydantic's smart_deepcopy
+                # and always using a default_factory instead of a default
+                if default is not UNSET:
+                    if default_factory is not UNSET:
+                        raise RuntimeError(
+                            "Both default and default_factory were defined! thats bad!!"
+                        )
```

Comment:
Can you create a custom exception for this? 😊

Reply:
oops, i should probably write a test for this as well